website/content/en/docs/building-operators/golang/webhook.md

```diff
@@ -0,0 +1,185 @@
+Kubebuilder also has a guide that walks through implementing webhooks for their example `CronJob` resource.
+
+### Create Validation Webhook
+
+As an example, let's start by creating a validation webhook.
+
+```sh
+$ operator-sdk create webhook --group cache --version v1alpha1 --kind Memcached --defaulting --programmatic-validation
+```
+
+After, `create webhook` command, the following message will appear on the terminal. It scaffolds out `api/<version>/<kind>_webhook.go` file. In this example, it would be `api/v1alpha1/memcached_webhook.go`.
+
+```sh
+Writing kustomize manifests for you to edit...
+Writing scaffold for you to edit...
+api/v1alpha1/memcached_webhook.go
+```
+
+The `--defaulting` flag will scaffold the resources required for a mutating webhook, and the `--programmatic-validation` flag will scaffold the resources required for a validating webhook. 
+In this case we scaffolded both.
+
+After running the `create webhook` command the file structure would be:
+
+```sh
+├── Dockerfile
+├── Makefile
+├── PROJECT
+├── api
+│   └── v1alpha1
+│       ├── memcached_webhook.go
+│       ├── webhook_suite_test.go
+├── config
+│   ├── certmanager
+│   │   ├── certificate.yaml
+│   │   ├── kustomization.yaml
+│   │   └── kustomizeconfig.yaml
+│   ├── default
+│   │   ├── manager_webhook_patch.yaml
+│   │   └── webhookcainjection_patch.yaml
+│   └── webhook
+│       ├── kustomization.yaml
+│       ├── kustomizeconfig.yaml
+│       └── service.yaml
+├── go.mod
+├── go.sum
+└── main.go
+```
+
+The scaffolded file `api/v1alpha1/memcached_webhook.go` has method signatures which need to be implemented for the validation webhook.
+
+Following this, there are a few steps which need to be done in your operator project to enable webhhoks. This will involve:
+
+1. Implementing the required methods for Validating or Mutating webhook in <kind>_webhook.go. An example of such implementation is provided [here](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html).
+
+2. Uncommenting sections in config/default/kustomization.yaml to enable webhook and cert-manager configuration through kustomize. Cert-manager (or any third party solution) can be used to provision certificates for webhook server. This is explained in detail here.
+
+### Generate webhook manifests and enable webhook deployment 
+
+Once your webhooks are implemented, all that’s left is to create the WebhookConfiguration manifests required to register your webhooks with Kubernetes:
+
+```sh
+$ make manifests
+```
+
+You will need to enable cert-manager and webhook deployment in order to deploy these webhooks properly. 
+To do so, edit the `config/default/kustomize.yaml` and uncomment the sections marked by `[WEBHOOK]` and `[CERTMANAGER]` comments. 
+More detail on this step can be found in the [Kubebuilder documentation](https://book.kubebuilder.io/cronjob-tutorial/running-webhook.html).
+
+### Update main.go so that running locally works 
```

Comment:
```suggestion

```
This section is not needed either. There is nothing to do in `main.go` the scaffolding will itself register the webhook in `main.go`.

Reply:
Modified.